In [1]:
import raw_utils as util
import os

import pandas as pd
import numpy as np

import random
random.seed(1746)

In [2]:
# Paths
cwd = os.getcwd()
csv_path = os.path.join(cwd, 'data/csv/')

# Filenames
nazario_csv = 'nazario_recent.csv'

## Phishing

First, read the csv with the recent emails.

In [3]:
phishing_text_raw = pd.read_csv(os.path.join(csv_path, nazario_csv), index_col=0, encoding='latin-1', dtype={'Body': 'object'})

In [4]:
phishing_text_raw.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 1916 entries, 0 to 1915
Data columns (total 1 columns):
 #   Column  Non-Null Count  Dtype 
---  ------  --------------  ----- 
 0   Body    1875 non-null   object
dtypes: object(1)
memory usage: 29.9+ KB


### Cleanup

#### Remove Useless Rows

We see that there are some rows with no body text. These were emails that contained only attachments, so it is safe to drop them.

In [5]:
phishing_text = phishing_text_raw.dropna()

Afterwards, we can see that there are some computer generated messages at the beginning of the mbox files, which we also need to remove.

In [9]:
phishing_text = phishing_text[phishing_text["Body"].str.contains("This text is part of the internal format of your mail folder, and is not\na real message.") == False]

In [10]:
print(phishing_text.shape)

(1869, 1)


Finally, we will remove the duplicate messages.

In [11]:
phishing_text = phishing_text[phishing_text.duplicated(keep='first') == False]

In [12]:
print(phishing_text.shape)

(1720, 1)


We see that we have 1720 emails to work with.